<h2>PROJECT CITYBIKE LUBLIN 

<span style="color:red">We don't have complete data, so we have to do analysis two times. First we will compare city bike in Lublin in 2016 (March, April, May) and 2017(March, April, May), and next 2014(October, November) and 2016(October, November).

In [1]:
import pandas as pd
import numpy as np
import datetime
import matplotlib.pyplot as plt
import plotly.graph_objects as go
import plotly.express as px

%matplotlib inline

<h1>1. FIRST PART OF ANALYSIS (2016 vs 2017)

In [2]:
c_b17 = pd.read_csv('c_b17')

c_b17.sample(3)

Unnamed: 0  bike_name           start_time             end_time  \
77450        84364      69153  2017-04-22 10:20:40  2017-04-22 10:28:39   
153677      166918      69790  2017-05-23 11:46:00  2017-05-23 11:53:00   
125984      136739      69536  2017-05-16 17:41:00  2017-05-16 17:56:00   

        start_place_id  end_place_id  start_day  start_month  day_of_month  \
77450           315238        673845        112            4            22   
153677          315270        315251        143            5            23   
125984          315228        315270        136            5            16   

        day_of_week                   duration  duration_sec  duration_min  \
77450             5  0 days 00:07:59.000000000           479      7.983333   
153677            1  0 days 00:07:00.000000000           420      7.000000   
125984            1  0 days 00:15:00.000000000           900     15.000000   

        duration_hour        date  start_hour  
77450        0.133056  2017-04-22          10  
153677       0.116667  2017-05-23          11  
125984       0.250000  2017-05-16          17

In [3]:
city_bike16 = pd.read_csv('c_b16')

#get only part of data frame which include information about data from March to May

c_b16_march = city_bike16[city_bike16['start_month']==3]

c_b16_april = city_bike16[city_bike16['start_month']==4]

c_b16_may = city_bike16[city_bike16['start_month']==5]

In [4]:
c_b16 = pd.concat(objs=[c_b16_march,c_b16_april,c_b16_may], ignore_index=True)

c_b16.sample(3)

Unnamed: 0  bike_name           start_time             end_time  \
112962      122508      69471  2016-04-22 17:18:16  2016-04-22 17:46:59   
115991      125829      69616  2016-04-23 14:45:00  2016-04-23 14:51:57   
212135      230848      69751  2016-05-14 17:29:00  2016-05-14 17:41:00   

        start_place_id  end_place_id  start_day  start_month  day_of_month  \
112962          674311        673833        113            4            22   
115991          315228        674333        114            4            23   
212135          315259        478054        135            5            14   

        day_of_week                   duration  duration_sec  duration_min  \
112962            4  0 days 00:28:43.000000000          1723     28.716667   
115991            5  0 days 00:06:57.000000000           417      6.950000   
212135            5  0 days 00:12:00.000000000           720     12.000000   

        duration_hour        date  start_hour  
112962       0.478611  2016-04-22          17  
115991       0.115833  2016-04-23          14  
212135       0.200000  2016-05-14          17

In [5]:
#For how many days were the bikes available for rental?

num_of_days = c_b16['start_day'].nunique()

print(f'The bikes were available for rental for {num_of_days} dyas in 2016')

num_of_days = c_b17['start_day'].nunique()

print(f'The bikes were available for rental for {num_of_days} dyas in 2017')

The bikes were available for rental for 72 dyas in 2016
The bikes were available for rental for 72 dyas in 2017


In [6]:
#How many rentals and bikes were there?

num_of_bikes = c_b16['bike_name'].nunique()

num_of_rent = c_b16['bike_name'].count()

print(f'Count of rentals:{num_of_rent}, count of bikes: {num_of_bikes} in 2016')

num_of_bikes = c_b17['bike_name'].nunique()

num_of_rent = c_b17['bike_name'].count()

print(f'Count of rentals:{num_of_rent}, count of bikes: {num_of_bikes} in 2017')

Count of rentals:304238, count of bikes: 891 in 2016
Count of rentals:185291, count of bikes: 850 in 2017


In [7]:
#The average time of a rental:

avg_time = c_b16['duration_min'].mean()

print(f'The average time of rental: {avg_time} in 2016')

avg_time = c_b17['duration_min'].mean()

print(f'The average time of rental: {avg_time} in 2017')

The average time of rental: 18.24584837090261 in 2016
The average time of rental: 17.7577186695522 in 2017


In [8]:
#Rentals under 20 minutes

rentals_under_20 = len(c_b16[c_b16['duration_min']<20])

print('Num of rentals under 20 minutes in 2016:', rentals_under_20)

rentals_under_20 = len(c_b17[c_b17['duration_min']<20])

print('Num of rentals under 20 minutes in 2016:', rentals_under_20)

Num of rentals under 20 minutes in 2016: 239639
Num of rentals under 20 minutes in 2016: 148309


In [9]:
#Most popular month


month = {3:'March',4:'April',5:'May'}

top_month = c_b16.groupby('start_month').agg({'bike_name':'count'}).sort_values(by='bike_name', ascending=False).iloc[0].name

print(f'The most popular month is: {month[top_month]} /2016')

c_b16.groupby('start_month').agg({'bike_name':'count'}).sort_values(by=['bike_name'], ascending=False)

top_month = c_b17.groupby('start_month').agg({'bike_name':'count'}).sort_values(by='bike_name', ascending=False).iloc[0].name

print(f'The most popular month is: {month[top_month]} /2017')


The most popular month is: May /2016
The most popular month is: May /2017


In [10]:
#Daily top score

top_day = c_b16.groupby('date').agg({'bike_name':'count'}).sort_values(by='bike_name', ascending=False).iloc[0].name

print(f'The most popular day: {top_day}')

top_day = c_b17.groupby('date').agg({'bike_name':'count'}).sort_values(by='bike_name', ascending=False).iloc[0].name

print(f'The most popular day: {top_day}')


The most popular day: 2016-04-05
The most popular day: 2017-04-02


In [11]:
#Most popular start station

top_s_station = c_b16['start_place_id'].value_counts().head(1).index[0]

print(f'The most popular start station number: {top_s_station} in 2016')

top_s_station = c_b17['start_place_id'].value_counts().head(1).index[0]

print(f'The most popular start station number: {top_s_station} in 2017')

The most popular start station number: 315269 in 2016
The most popular start station number: 315269 in 2017


In [12]:
#Most popular end station

top_end_station = c_b16['end_place_id'].value_counts().head(1).index[0]

print(f'The most popular start station number: {top_end_station} in 2016')

top_end_station = c_b17['end_place_id'].value_counts().head(1).index[0]

print(f'The most popular start station number: {top_end_station} in 2017')

The most popular start station number: 315269 in 2016
The most popular start station number: 315269 in 2017


In [13]:
#How often the bikes were rented

#2016 

freq_2016 = 22032000 // 783692

print(f'The bikes were rented every {freq_2016} sec. in 2016')

#2017

freq_2017 = 6220800 // 185291

print(f'The bikes were rented every {freq_2017} sec. in 2017')

The bikes were rented every 28 sec. in 2016
The bikes were rented every 33 sec. in 2017


In [14]:
#Create a plot for average rent number per month

#2016

average_rent_16 = c_b16.groupby('start_month').agg({'bike_name':'count'}) #I count num of rent per month

average_rent_16['num_day_in_month'] = c_b16.groupby(['start_month'])[['day_of_month']].nunique() #I count num of days in month

average_rent_16['result'] = average_rent_16['bike_name']//average_rent_16['num_day_in_month'] #I count average rent number for every month

print('Average rent number per month in 2016\n',average_rent_16,'\n\n')

#2017

average_rent_17 = c_b17.groupby('start_month').agg({'bike_name':'count'}) 

average_rent_17['num_day_in_month'] = c_b17.groupby(['start_month'])[['day_of_month']].nunique() 

average_rent_17['result'] = average_rent_17['bike_name']//average_rent_17['num_day_in_month'] 

print('Average rent number per month in 2017\n',average_rent_17)

Average rent number per month in 2016
              bike_name  num_day_in_month  result
start_month                                     
3                15941                11    1449
4               131390                30    4379
5               156907                31    5061 


Average rent number per month in 2017
              bike_name  num_day_in_month  result
start_month                                     
3                25909                11    2355
4                61829                30    2060
5                97553                31    3146


In [15]:
d = {'2016':average_rent_16['result'].to_list(), '2017':average_rent_17['result'].to_list()}

df = pd.DataFrame(data=d, index=['March','April','May'])

df

2016  2017
March  1449  2355
April  4379  2060
May    5061  3146

In [ ]:
df.plot(y = ['2016','2017'],use_index=True, kind='bar',rot=360, title = 'Average rent number per month', legend = True);

In [ ]:
#Create a plot for average rent number per day of week in 2016

rent_num_16 = c_b16.groupby('day_of_week').agg({'bike_name':['count']}) #I count num of rentals per day of week

num_of_day_df = pd.DataFrame(c_b16.groupby(['date'])['day_of_week'].unique().value_counts()).sort_index()

rent_num_16['num_of_days']=num_of_day_df['day_of_week']

rent_num_16['num_per_day'] = rent_num_16['bike_name']['count'] / rent_num_16['num_of_days']

rent_num_16.round(2)

In [ ]:
#Create a plot for average rent number per day of week in 2017

rent_num_17 = c_b17.groupby('day_of_week').agg({'bike_name':['count']}) #I count num of rentals per day of week

num_of_day_df = pd.DataFrame(c_b17.groupby(['date'])['day_of_week'].unique().value_counts()).sort_index()

rent_num_17['num_of_days']=num_of_day_df['day_of_week']

rent_num_17['num_per_day'] = rent_num_17['bike_name']['count'] / rent_num_17['num_of_days']

rent_num_17.round(2)

In [ ]:
d = {'2016':rent_num_16['num_per_day'].to_list(), '2017':rent_num_17['num_per_day'].to_list()}

df = pd.DataFrame(data=d, index=['Monday','Tuesday','Wednesday','Thuersday','Friday','Saturday','Sunday'])

df

In [ ]:
df.plot(y = ['2016','2017'],kind='line',rot=360, figsize = (10,7),title = 'Average rent number per day of week', legend = True);

In [ ]:
#Create a plot for average time of rent per day of week

ave_time_16 = c_b16.groupby('day_of_week').agg({'duration_min':'mean'})
ave_time_16.reset_index(inplace=True)
ave_time_16

In [ ]:
ave_time_17 = c_b17.groupby('day_of_week').agg({'duration_min':'mean'})
ave_time_17.reset_index(inplace=True)
ave_time_17

In [ ]:
d={'2016': ave_time_16['duration_min'].to_list(), '2017':ave_time_17['duration_min'].to_list()}

df = pd.DataFrame(data=d, index=['Monday','Tuesday','Wednesday','Thuersday','Friday','Saturday','Sunday'])

df

In [ ]:
df.plot(y = ['2016','2017'],kind='line',rot=360, figsize = (10,7),title = 'Average time', legend = True);

In [ ]:
#Create a plot for average number of rents per hour 

ave_rent_per_h_16= c_b16.groupby(['start_hour']).agg({'bike_name':'count'})

ave_rent_per_h_16.head()

In [ ]:
ave_rent_per_h_17= c_b17.groupby(['start_hour']).agg({'bike_name':'count'})

ave_rent_per_h_17.head()

In [ ]:
d = {'2016':ave_rent_per_h_16['bike_name'].to_list() ,'2017':ave_rent_per_h_17['bike_name'].to_list()}

df = pd.DataFrame(data=d)

df.head()

In [ ]:
df.plot(y = ['2016','2017'],kind ='line', figsize = (15,7), use_index=True, xticks = (range(24)),rot = 45, grid=True);
                   


<h1>CONCLUSIONS

<span style="color:red">1.I compare first 3 months from 2016 and 2017. The bikes were available by 72 days.
    
<span style="color:red">2.In 2016 were more available bikes than 2017(891 vs 850). It was the one reasons why in 2016 were more rentals than 2017 (304238 vs 185291). The second reason could by the weather. In 2017 average day's temperature was lower than 2016 at the same period.
    
<span style="color:red">3.The average time of rental was similar in 2016 and 2017 (18.24 vs 17.75).

<span style="color:red">4.Rentals under 20 minutes are 78% all rentals in 2016 and 80% in 2017.

<span style="color:red">5.The most popular month in both year was May.

<span style="color:red">6.The most popular day in 2016 was 5th April - it was on Tuesday, very sunny day. In 2017 top day was 2th April, on very sunny Sunday.
    
<span style="color:red">7.The most popular start and end station is the same in both cases - station 315269
    
<span style="color:red">8.The bikes were rented every 28 sec in 2016 and every 33 sec in 2017.
    
<span style="color:red">9.The average rent number per month was higer in 2016 in April and May but lower in March. We can notice that num of rent in 2017 is similar in every month. In 2016 in March were definitely fewer rentals than April and May - upward trend
 
<span style="color:red">10.The average num rent per day of week in 2016 and 2017 is higer in the week tha at the weekend. The most popular day in 2016 was Thuersday and in 2017 Monday. 

<span style="color:red">11.The average time of rent is higer at the weekend than the week. People have more time to go for a bike ride at the weekend. In the week bikes are usefull for travel to job or school so rentals are shorter.
    
<span style="color:red">12.The average number of rents per hour is very similar in 2016 and 2017. We can see increase num of rent beetwen 7 and 8 a.m and 3 and 4 p.m - we can assume that for many people the bicycle is a kind of transport to and from work and school 
    
<span style="color:red">

<h1>2. SECOND PART OF ANALYSIS (2016 vs 2014)

In [ ]:
c_b14 = pd.read_csv('c_b14')

#get only October and Nobember (in Septeber were only 12 results)

c_b14 = c_b14[c_b14['start_month']!=9]

c_b14.head()

In [ ]:
city_bike16 = pd.read_csv('c_b16')

#get only part of data frame which include information about data from Octpober and November

c_b16_oct = city_bike16[city_bike16['start_month']==10]

c_b16_nov = city_bike16[city_bike16['start_month']==11]

In [ ]:
c_b16 = pd.concat(objs=[c_b16_oct,c_b16_nov], ignore_index=True)

c_b16.sample(3)

In [ ]:
#For how many days were the bikes available for rental?

num_of_days = c_b14['start_day'].nunique()

print(f'The bikes were available for rental for {num_of_days} dyas in 2014')

num_of_days = c_b16['start_day'].nunique()

print(f'The bikes were available for rental for {num_of_days} dyas in 2016')

In [ ]:
#How many rentals were there?


num_of_rent = c_b14['start_time'].count()

print(f'Count of rentals:{num_of_rent} in 2014')

num_of_rent = c_b16['bike_name'].count()

print(f'Count of rentals:{num_of_rent} in 2016')

In [ ]:
#The average time of a rental:

avg_time = c_b14['duration_min'].mean()

print(f'The average time of rental: {avg_time} in 2014')

avg_time = c_b16['duration_min'].mean()

print(f'The average time of rental: {avg_time} in 2016')

In [ ]:
#Rentals under 20 minutes

rentals_under_20 = len(c_b14[c_b14['duration_min']<20])

print('Num of rentals under 20 minutes in 2014:', rentals_under_20)

rentals_under_20 = len(c_b16[c_b16['duration_min']<20])

print('Num of rentals under 20 minutes in 2016:', rentals_under_20)

In [ ]:
#Number of rentals per month

c_b14.groupby('start_month').agg({'start_time':'count'}).sort_values(by=['start_time'], ascending=False)

In [ ]:
c_b16.groupby('start_month').agg({'start_time':'count'}).sort_values(by=['start_time'], ascending=False)

In [ ]:
#Daily top score in September-November


top_day = c_b14.groupby('date').agg({'start_time':'count'}).sort_values(by='start_time', ascending=False).iloc[0].name

print(f'The most popular day: {top_day}')

top_day = c_b16.groupby('date').agg({'bike_name':'count'}).sort_values(by='bike_name', ascending=False).iloc[0].name

print(f'The most popular day: {top_day}')



In [ ]:
#How often the bikes were rented

#Num of day * num of hours per day * num of minutes per hour * num of seconds per minute // 103704 rent per 62 days:

#62 * 24 * 60 * 60 = 5356800‬

freq = 5356800 // 103704

print(f'The bikes were rented every {freq} sec. in 2014')

#2016 

freq_2016 = 22032000 // 783692

print(f'The bikes were rented every {freq_2016} sec. in 2016')

In [ ]:
#Create a plot for average rent number per month

#2014

average_rent_14 = c_b14.groupby('start_month').agg({'start_time':'count'}) #I count num of rent per month

average_rent_14['num_day_in_month'] = c_b14.groupby(['start_month'])[['day_of_month']].nunique() #I count num of days in month

average_rent_14['result'] = average_rent_14['start_time']//average_rent_14['num_day_in_month'] #I count average rent number for every month

print('Average rent number per month in 2014\n',average_rent_14,'\n\n')

#2016

average_rent_16 = c_b16.groupby('start_month').agg({'bike_name':'count'}) #I count num of rent per month

average_rent_16['num_day_in_month'] = c_b16.groupby(['start_month'])[['day_of_month']].nunique() #I count num of days in month

average_rent_16['result'] = average_rent_16['bike_name']//average_rent_16['num_day_in_month'] #I count average rent number for every month

print('Average rent number per month in 2016\n',average_rent_16,'\n\n')

In [ ]:
d = {'2014':average_rent_14['result'].to_list(), '2016':average_rent_16['result'].to_list()}

df = pd.DataFrame(data=d, index=['October','November'])

df

In [ ]:
trace_one = go.Bar(x=df.index, 
    y = df['2014'],
            name='2014',
                  marker=dict(color='rgba(199,0,199,1)'))

trace_two = go.Bar(x=df.index, 
    y = df['2016'],
            name='2016',
                marker=dict(color='rgba(120,120,120,1)')  )

data=[trace_one,trace_two]

layout = { 'barmode':'group',
            'xaxis':{'title':'Month'},
            'yaxis':{'title':'Num of rent'}}


fig = go.Figure(data, layout=layout)

fig.update_layout(
    title={'text':'Average rent number per month',
           'font': {
              'family': 'Calibri',
              'size': 20
        },'y':0.95,
        'x':0.5,
        'xanchor': 'center',
        'yanchor': 'top'})

fig.show()

In [ ]:
#Create a plot for average rent number per day of week in 2014

rent_num_14 = c_b14.groupby('day_of_week').agg({'start_time':['count']}) #I count num of rentals per day of week

num_of_day_df = pd.DataFrame(c_b14.groupby(['date'])['day_of_week'].unique().value_counts()).sort_index()

rent_num_14['num_of_days']=num_of_day_df['day_of_week']

rent_num_14['num_per_day'] = rent_num_14['start_time']['count'] / rent_num_14['num_of_days']

rent_num_14.round(2)

In [ ]:
#Create a plot for average rent number per day of week in 2016

rent_num_16 = c_b16.groupby('day_of_week').agg({'bike_name':['count']}) #I count num of rentals per day of week

num_of_day_df = pd.DataFrame(c_b16.groupby(['date'])['day_of_week'].unique().value_counts()).sort_index()

rent_num_16['num_of_days']=num_of_day_df['day_of_week']

rent_num_16['num_per_day'] = rent_num_16['bike_name']['count'] / rent_num_16['num_of_days']

rent_num_16.round(2)

In [ ]:
d = {'2014':rent_num_14['num_per_day'].to_list(), '2016':rent_num_16['num_per_day'].to_list()}

df = pd.DataFrame(data=d, index=['Monday','Tuesday','Wednesday','Thuersday','Friday','Saturday','Sunday'])

df

In [ ]:
trace_one = go.Bar(x=df.index, 
            y = df['2014'],
            name='2014',
            marker=dict(color='rgba(199,0,199,1)'))

trace_two = go.Bar(x=df.index, 
            y = df['2016'],
            name='2016',
            marker=dict(color='rgba(120,120,120,1)'))

data=[trace_one,trace_two]

layout = { 'barmode':'group',
            'xaxis':{'title':'Days'},
            'yaxis':{'title':'Num of rent'}}


fig = go.Figure(data, layout=layout)

fig.update_layout(
    title={'text':'Average rent number per day of week',
           'font': {
              'family': 'Calibri',
              'size': 20
        },'y':0.95,
        'x':0.5,
        'xanchor': 'center',
        'yanchor': 'top'})


fig.show()

In [ ]:
#Create a plot for average time of rent per day of week

#2014

ave_time_14 = c_b14.groupby('day_of_week').agg({'duration_min':'mean'})
ave_time_14.reset_index(inplace=True)
ave_time_14

In [ ]:
#2016

ave_time_16 = c_b16.groupby('day_of_week').agg({'duration_min':'mean'})
ave_time_16.reset_index(inplace=True)
ave_time_16

In [ ]:
d={'2014': ave_time_14['duration_min'].to_list(), '2016':ave_time_16['duration_min'].to_list()}

df = pd.DataFrame(data=d, index=['Monday','Tuesday','Wednesday','Thuersday','Friday','Saturday','Sunday'])

df

In [ ]:
'''trace_one = go.Bar(x=df.index, 
            y = df['2014'],
            name='2014',
            marker=dict(color='rgba(199,0,199,1)'))

trace_two = go.Bar(x=df.index, 
            y= df['2016'],
            name='2016',
            marker=dict(color='rgba(120,120,120,1)'))

data=[trace_one,trace_two]

layout = { 'barmode':'group',
            'xaxis':{'title':'Time'},
            'yaxis':{'title':'Num of rent'}}


fig = go.Figure(data, layout=layout)

fig.update_layout(
    title={'text':'Average time of rent number per day of week',
           'font': {
              'family': 'Calibri',
              'size': 20
        },'y':0.95,
        'x':0.5,
        'xanchor': 'center',
        'yanchor': 'top'})


fig.show()'''


trace_one = go.Bar(x=df.index, 
    y = df['2014'],
            name='2014',
                  marker=dict(color='rgba(199,0,199,1)'))

trace_two = go.Bar(x=df.index, 
    y = df['2016'],
            name='2016',
                marker=dict(color='rgba(120,120,120,1)')  )

data=[trace_one,trace_two]

layout = { 'barmode':'group',
            'xaxis':{'title':'Days'},
            'yaxis':{'title':'Time of rent'}}


fig = go.Figure(data, layout=layout)

fig.update_layout(
    title={'text':'Average time',
           'font': {
              'family': 'Calibri',
              'size': 20
        },'y':0.95,
        'x':0.5,
        'xanchor': 'center',
        'yanchor': 'top'})

fig.show()

In [ ]:
#Create a plot for average number of rents per hour 

#2014

ave_rent_per_h_14= c_b14.groupby(['start_hour']).agg({'start_time':'count'})

ave_rent_per_h_14.head()

In [ ]:
#2016

ave_rent_per_h_16= c_b16.groupby(['start_hour']).agg({'bike_name':'count'})

ave_rent_per_h_16.head()

In [ ]:
d = {'2014':ave_rent_per_h_14['start_time'].to_list() ,'2016':ave_rent_per_h_16['bike_name'].to_list()}

df = pd.DataFrame(data=d)

df.head()

In [ ]:
trace_one = go.Bar(x=df.index, 
            y = df['2014'],
            name='2014',
            marker=dict(color='rgba(199,0,199,1)'))

trace_two = go.Bar(x=df.index, 
            y = df['2016'],
            name='2016',
            marker=dict(color='rgba(120,120,120,1)'))

data=[trace_one,trace_two]

layout = { 'barmode':'group',
            'xaxis':{'title':'Hours'},
            'yaxis':{'title':'Num of rent'}}


fig = go.Figure(data, layout=layout)

fig.update_layout(
    title={'text':'Average rent number per hour',
           'font': {
              'family': 'Calibri',
              'size': 20
        },'y':0.95,
        'x':0.5,
        'xanchor': 'center',
        'yanchor': 'top'})



fig.show()

<h1>CONCLUSIONS

<span style="color:red">1.I compare first 2 months from 2014 and 2016. The bikes were available by 61 days.
    
<span style="color:red">2.In 2014 were more rentals than 2016 (103692 vs 55310). 
    
<span style="color:red">3.The average time of rental:18.77 in 2014 and 14.29 in 2016.

<span style="color:red">4.Rentals under 20 minutes are 77% all rentals in 2014 and 88% in 2016.

<span style="color:red">5.The most popular day in 2014 was 13th October - it was on Monday. In 2016 top day was 2th October, on Sunday.
    
<span style="color:red">6.The bikes were rented every 51 sec in 2014 and every 28 sec in 2016.
    
<span style="color:red">7.The average rent number per month was higer in 2014. We can notice that num of rent in October is two times the size of November. It is probably becouse the weather was beter in October.
 
<span style="color:red">8.The average num rent per day of week in 2014 and 2016 is higer in the week than at the weekend. The most popular day in 2014 was Wednesday and in 2016 Thuersday. 

<span style="color:red">9.The average time of rent is higer at the weekend than the week. People have more time to go for a bike ride at the weekend. In the week bikes are usefull for travel to job or school so rentals are shorter.
    
<span style="color:red">10.The average number of rents per hour is very similar in 2014 and 2016. We can see increase num of rent beetwen 7 and 8 a.m and 3 and 4 p.m - we can assume that for many people the bicycle is a kind of transport to and from work and school 
   